e2e/nomostest/gitproviders/git-provider.go

```diff
@@ -63,3 +63,17 @@ func NewGitProvider(t testing.NTB, provider string, logger *testlogger.TestLogge
 		return &LocalProvider{}
 	}
 }
+
+// WithLogger updates an existing GitProvider to use the provider TestLogger
+func WithLogger(t testing.NTB, provider GitProvider, logger *testlogger.TestLogger) GitProvider {
+	switch p := provider.(type) {
+	case *BitbucketClient:
```

Comment:
why only change the logger for bitbucket?

Reply:
it's the only one that configures a logger at the moment